qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -69,35 +70,65 @@ def __init__(
         self._properties = None
         self._defaults = None
 
-    def run(self, qobj: Qobj, job_name: Optional[str] = None) -> IBMQJob:
+    def run(
+            self,
+            qobj: Qobj,
+            job_name: Optional[str] = None,
+            job_share_level: Optional[str] = None
+    ) -> IBMQJob:
         """Run a Qobj asynchronously.
 
         Args:
             qobj: description of job.
             job_name: custom name to be assigned to the job. This job
                 name can subsequently be used as a filter in the
                 ``jobs()`` function call. Job names do not need to be unique.
+            job_share_level: allows sharing a job at the hub/group/project and
+                global level. The possible job share levels are: "global", "hub",
+                "group", "project", and "none".
+                    * global: the job is public to any user.
+                    * hub: the job is shared between the users in the same hub.
+                    * group: the job is shared between the users in the same group.
+                    * project: the job is shared between the users in the same project.
+                    * none: the job is not shared at any level.
+                If the job share level is not specified, then the job is not shared at any level.
 
         Returns:
             an instance derived from BaseJob
 
         Raises:
             SchemaValidationError: If the job validation fails.
             IBMQBackendError: If an unexpected error occurred while submitting
-                the job.
+                the job or the job share level is not valid.
         """
         # pylint: disable=arguments-differ
+        api_job_share_level = None
+        if job_share_level:
+            try:
+                api_job_share_level = ApiJobShareLevel(job_share_level)
+            except ValueError:
+                raise IBMQBackendError(
```

Comment:
Can you raise a different exception here? I'd prefer if we separate exceptions that the user cannot control (e.g. job submit failed due to server error) from the ones that the user can fix.

Reply:
Yes, that would be a good rule to follow. I've seen `IBMQBackendValueError` used elsewhere and it appears to be more suited for this use case.